test/ShapeCrawler.Tests.Unit/ShapeTests.cs

```diff
@@ -672,4 +672,50 @@ public void CornerSize_setter_sets_values()
         shape.CornerSize.Should().Be(expected);
         pres.Validate();
     }
+    
+    [Test]
+    public void Name_Setter_sets_shape_name()
+    {
+        // Arrange
+        var pptx = StreamOf("006_1 slides.pptx");
+        var pres = new Presentation(pptx);
+        var stream = new MemoryStream();
+        var shape = pres.Slides[0].Shapes.GetByName("Shape 1");
+
+        // Act
+        shape.Name = "New Name";
+
+        // Assert
+        pres.SaveAs(stream);
+        pres = new Presentation(stream);
+        shape = pres.Slides[0].Shapes.GetByName("New Name");
+        shape.Name.Should().Be("New Name");
+        pres.Validate();
+    }
+    
+    [Test]
+    public void Name_Setter_sets_grouped_shape_name()
+    {
+        // Arrange
+        var pptx = StreamOf("autoshape-grouping.pptx");
+        var pres = new Presentation(pptx);
+        var stream = new MemoryStream();
+        var groupShape = pres.Slides[0].Shapes.GetByName<IGroupShape>("Group 2");
+        var shape1 = groupShape.Shapes.GetByName("Shape 1");
+        var shape2 = groupShape.Shapes.GetByName("Shape 2");
```

Comment:
The initial values of `shape1` and `shape2` are not used. It looks like these statements on 704 and 705 lines can be removed?

@blockingHD, thank you for the Pull Request!

Reply:
No worries, will remove them. The logic was that it checks only the group name is updated and it doesn't do some random cascading update.